server/bin/pytests/test_logger_type.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os, sys
+import logging
+import pytest
+
+from pbench import PbenchConfig, BadConfig, get_pbench_logger
+
+_NAME_ = "pbench-logger-test"
+cfg_name = os.environ["CONFIG"]
+logdir = os.environ["LOGSDIR"]
+
+log_files = {
+    "file": "file.log",
+    "devlog": "devlog.log",
+    "hostport": "hostport.log"
+    }
+
+log_msgs = {
+    "file": "logger_type=file in file file.log",
+    "devlog": "logger_type=devlog in file devlog.log",
+    "hostport": "logger_type=hostport in file hostport.log"
+}
+
+def mock_the_handler(logger, logger_type, fname):
+    
+    # Assumption: only one Handler is present.
+    hdlr = logger.logger.handlers[0]
+    logger.logger.removeHandler(hdlr)
+
+    # logger.logger is used: the first logger is used to format 
+    # the logs with the help of _styleAdapter and the second is 
+    # used to log the messages
+    fh = logging.FileHandler(os.path.join(logdir,fname))
+    fh.setLevel(logging.DEBUG)
+    logger.logger.addHandler(fh)
+    
+    return logger
+
+def test_pbench_logger():
+
+    config = PbenchConfig(cfg_name)
+    logger = get_pbench_logger(_NAME_, config)
+
+    logger_type = config.get("logging", "logger_type")
+
+    if(logger_type=='file' or logger_type=='devlog' or logger_type=='hostport'):
```

Comment:
... or (shorter and a bit more readable):

          if logger_type in ['file', 'devlog', 'hostport']:
                ....
or even:

          if logger_type in log_files.keys():
               ....

although this last one is probably too cryptic.